src/modules/signer/signer.service.ts

```diff
@@ -354,6 +345,19 @@ export class SignerService {
     return sig;
   }
 
+  /**
+   * Signs message with whatever signing implementation underlying signer provides. Unlike `signMessage` this method does not guaraties that signature will conform to EIP-191
+   * @param message Stringified message
+   */
+  async sign(message: string): Promise<string> {
+    const messageBytes = Buffer.from(message);
+    const signature = await this.signer.signMessage(messageBytes);
+    if (!this._publicKey || this._isEthSigner === undefined) {
+      await this._determineSigner(messageBytes, signature);
+    }
+    return signature;
+  }
```

Comment:
I don't get the "this method does not guaraties that signature will conform to EIP-191" as `signer.signMessage` performs an EIP-191 signature, right? 

https://docs.ethers.org/v5/api/signer/#Signer-signMessage
> A signed message is prefixd with "\x19Ethereum Signed Message:\n" and the length of the message, using the [hashMessage](https://docs.ethers.org/v5/api/utils/hashing/#utils-hashMessage) method, so that it is [EIP-191](https://eips.ethereum.org/EIPS/eip-191) compliant. If recovering the address in Solidity, this prefix will be required to create a matching hash.

Reply:
I think this is requirements to implementers of `Signer` interface. But not all implementations are conform to this. If I remember correctly this is the reason why we are prefixing message manually before passing it to Metamask signer https://github.com/energywebfoundation/iam-client-lib/blob/9d01b84fb1d9dd67971688b010811ad786050538/src/modules/signer/signer.service.ts#L347